src/actions/functionList.js

```diff
@@ -0,0 +1,56 @@
+//TODO: I don't like the filename... to many files with the same name
+
+
+// TODO move into a class and export methods to hide state
+let refreshTime = 350000
+let interval = null
+
+let intervalCounter = 0
+
+export function refreshFunctionList() {
+  return dispatch => {
+    dispatch({type:'FUNCTION_LIST_LOAD', status:'PENDING'})
+
+    fetch('/system/functions',
+    {
+        method: "GET", 
+        credentials: 'include'
+    })
+        .then(res => res.json())
+        .then(response => {
+            if (response && response.functions) {
+              dispatch({type:'FUNCTION_LIST_LOAD', status:'SUCCESS', storeList: response.functions})
+            }
+        }).catch(error => {
+          console.error(error)
+          dispatch({type:'FUNCTION_LIST_LOAD', status:'FAILED', message: error.message})
+        })
+  }
+}
+
+
+// TODO: Switch to timer to allow flexible timeouts?
+// As is, the timer can't change if we add a way to change it for the user.
```

Comment:
We could utilize an env var in the future, but this is existing functionality for the current gateway. 
Something that could be updated at another time. For now, let's just focus on UI and remove this comment please